pootle/apps/pootle_store/models.py

```diff
@@ -446,13 +445,17 @@ def save(self, *args, **kwargs):
                     self.state = UNTRANSLATED
                     self.store.mark_dirty(CachedMethods.WORDCOUNT_STATS)
 
-        # Updating unit from the .po file should not change its revision property,
+        # Updating unit from the .po file set its revision property to
+        # a new value (the same for all units during its store updated)
         # since that change doesn't require further sync but note that
         # auto_translated units require further sync
-        if ((not self._from_update_stores or self._auto_translated) and
-            (self._target_updated or self._state_updated
-             or self._comment_updated)):
-            self.revision = Revision.incr()
+        revision = kwargs.pop('revision', None)
+        if revision is not None and not self._auto_translated:
+            self.revision = revision
+        else:
+            if (self._target_updated or self._state_updated or
+                self._comment_updated):
```

Comment:
The following is simpler:

```python
        elif self._target_updated or self._state_updated or self._comment_updated:
```

Despite it exceeds the line length, I think this is a case where it is desirable to use the soft limit of 84 chars for the sake of readability.